plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,507 @@
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: gpg_keypair
+author: "Austin Lucas Lake (@austinlucaslake)"
+short_description: Generate or delete GPG private and public keys
+version_added: 2.20.0
+description:
+    - "This module allows one to generate or delete GPG private and public keys using GnuPG (gpg)."
+requirements:
+    - gpg >= 2.1
+attributes:
+    check_mode:
+        description:
+            - Can run in check_mode and return changed status prediction without modifying target.
+        support: full
+    diff_mode:
+        description:
+            - Will return details on what has changed (or possibly needs changing in check_mode), when in diff mode.
+        support: none
+options:
+    state:
+        description:
+            - Whether the private and public keys should exist or not, taking action if the state is different from what is stated.
+        type: str
+        default: present
+        choices: [ present, absent ]
+    key_type:
+        description:
+            - Specifies the type of key to create.
+        type: str
+        choices: [ 'RSA', 'DSA', 'ECDSA', 'EDDSA' ]
+    key_length:
+        description:
+            - For non-ECC keys, this specifies the number of bits in the key to create.
+            - For RSA keys, the minimum is V(1024), the maximum is V(4096), and the default is V(3072).
+            - For DSA keys, the minimum is V(768), the maximum is V(3072), and the default is V(2048).
+            - As per gpg's behavior, values below the allowed ranges will be set to the respective defaults, and values above will saturate at the maximum.
+        type: int
+    key_curve:
+        description:
+            - For ECC keys, this specifies the curve used to generate the keys.
+            - If O(key_type=EDDSA), O(key_curve=ed25519) is required.
+            - If O(key_curve=ed25519) is only supported if O(key_type=EDDSA).
+            - This is required if O(key_type=ECDSA) or O(key_type=EDDSA) and it is ignored if O(key_type=RSA) or O(key_type=DSA).
+        type: str
+        choices: [ 'nistp256', 'nistp384', 'nistp521', 'brainpoolP256r1', 'brainpoolP384r1', 'brainpoolP512r1', 'secp256k1', 'ed25519' ]
+    key_usage:
+        description:
+            - Specifies usage(s) for key.
+            - V(cert) is given to all primary keys regardess, however can be used to only give V(vert) usage to a key.
+            - If not usage is specified, all of valid usages for the given key type are assigned.
+            - O(key_usage=encr) is only supported is O(key_type=RSA).
+        type: list
+        elements: str
+        default: []
+        choices: [ 'encr', 'sign', 'auth', 'cert' ]
+    subkeys:
+        description:
+            - List of subkeys with their own respective key types, lengths, curves, and usages.
+        type: list
+        elements: dict
+        default: []
+        suboptions:
+            subkey_type:
```

Comment:
How about using
```suggestion
            key_type:
```
here (similar below?). That this is already a subkey is clear from this being part of `subkeys`.

Using the same keys as for the primary key makes it easy to copy over the values - you only have to change the indentation (except for the `-` before the first key for the subkey).

Reply:
Sounds good. Did that originally cause of name collision in documentation when using O(), but one of your other eliminates that issue.